ggshield/cmd/sca/scan/ci.py

```diff
@@ -64,15 +68,39 @@ def scan_ci_cmd(
             return output_handler.process_scan_all_result(scan)
 
         check_git_dir()
-        commit_count = len(
-            collect_commit_range_from_ci_env(config.user_config.verbose)[0]
-        )
+
+        new_commits, _ = collect_commit_range_from_ci_env(config.user_config.verbose)
+        previous_commit, _ = get_previous_commit_from_ci_env(config.user_config.verbose)
+
+        # If no sha provided, consider repo was empty before
+        if previous_commit == EMPTY_SHA or previous_commit == "null":
+            previous_commit = None
+
+        # If we fail to fetch current state, we set it to HEAD
+        if len(new_commits) == 0:
+            current_state = "HEAD"
+        else:
+            current_state = new_commits[-1]
+
+        if current_state == EMPTY_SHA or current_state == "null":
+            current_state = "HEAD"
+
         if config.user_config.verbose:
-            click.echo(f"Commits to scan: {commit_count}", err=True)
+            if len(new_commits) > 0:
+                click.echo("List of new commits: ")
+                for commit in new_commits:
+                    click.echo(f"   {commit}")
+
+            click.echo(
+                f"Comparing commit {current_state} to commit {previous_commit}",
+                err=True,
+            )
```

Comment:
This is exactly the same code for both verticals and could be factorized and done in git_hooks.

Reply:
I factorized it